libraries/rush-lib/src/utilities/WebClient.ts

```diff
@@ -53,49 +56,124 @@ export enum WebClientProxy {
   Detect,
   Fiddler
 }
+export interface IRequestOptions extends RequestOptions, Pick<IFetchOptionsWithBody, 'body' | 'redirect'> {}
+
+export type FetchFn = (
+  url: string,
+  options: IRequestOptions,
+  isRedirect?: boolean
+) => Promise<IWebClientResponse>;
+
+const makeRequestAsync: FetchFn = async (
+  url: string,
+  options: IRequestOptions,
+  redirected: boolean = false
+) => {
+  const { body, redirect } = options;
+
+  return await new Promise(
```

Comment:
I think the one thing that this didn't handle that I remember coding but not merging was GZipped responses. Those needed to be manually decoded. Not sure if that's something we want to include in this PR